pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TopicDoesNotExistsTest.java

```diff
@@ -61,11 +61,11 @@ public void testCreateProducerOnNotExistsTopic() throws PulsarClientException, I
                     .sendTimeout(100, TimeUnit.MILLISECONDS)
                     .create();
             Assert.fail("Create producer should failed while topic does not exists.");
-        } catch (PulsarClientException ignore) {
+        } catch (PulsarClientException e) {
+            Assert.assertTrue(e instanceof PulsarClientException.NotFoundException);
         }
         Thread.sleep(2000);
         HashedWheelTimer timer = (HashedWheelTimer) ((PulsarClientImpl) pulsarClient).timer();
-        Assert.assertEquals(timer.pendingTimeouts(), 0);
```

Comment:
Why are you removing this check?

Reply:
can you please add a Awaiatility loop that waits for the condition ?
 Assert.assertEquals(timer.pendingTimeouts(), 0);
 
 Awaiatility.await().untilAsserted(....)

Reply:
> Awaiatility.await().untilAsserted(....)

An awaiatility loop does not work, because the `sendTimeout` TimerTask will trigger another new TimerTask at the end of the task.